configure.ac

```diff
@@ -10,12 +14,225 @@ AM_INIT_AUTOMAKE(subdir-objects)
+LDFLAGS="$LDFLAGS -Wl,--no-as-needed -lc"
+
 # Checks for header files.
 AC_CHECK_HEADERS([altivec.h stddef.h stdint.h unistd.h stdio.h fenv.h float.h math.h])
 
+#############################################################################
+
+CNAME=`basename "$CC"`
+AM_CPPFLAGS="-DNDEBUG"
+AM_CFLAGS="-m64 -O3"
+
+## PowerPC-64 Specific options
+## Save CFLAGS
+SAVED_CFLAGS="$CFLAGS"
+
+ALTIVEC_CFLAGS="-mcpu=970 -maltivec"
+POWER_F128_CFLAGS="-mfloat128"
+POWER6_CFLAGS="-mcpu=power6 -maltivec"
+POWER7_CFLAGS="-mcpu=power7"
+POWER8_CFLAGS="-mcpu=power8"
+POWER9_CFLAGS="-mcpu=power9 -mfloat128"
+
+##### Compiler default includes ALTIVEC #####
+AC_SUBST([PVECLIB_DEFAULT_CFLAG], [""])
+
+CFLAGS=$AM_CFLAGS
+AC_MSG_CHECKING([if $CNAME default supports ALTIVEC])
+AC_LINK_IFELSE(
+	[AC_LANG_SOURCE([`cat testprograms/test_ppc_VMX.c`])],
+	[AC_MSG_RESULT([yes])]
+		AC_SUBST([PVECLIB_DEFAULT_CFLAG], [""]),
+	[AC_MSG_RESULT([no])]
+		AC_DEFINE([PVECLIB_DISABLE_ALTIVEC], [1], [Disable ALTIVEC])
+)
+
+if test "$retval" != "0"; then
+	AC_SUBST([PVECLIB_DEFAULT_CFLAG],[POWER8_CFLAGS])
+fi
+
+##### Compiler default includes DFP #####
+
+CFLAGS=$AM_CFLAGS
+AC_MSG_CHECKING([if $CNAME default supports DFP])
+AC_LINK_IFELSE(
+	[AC_LANG_SOURCE([`cat testprograms/test_ppc_DFP.c`])],
+	[AC_MSG_RESULT([yes])]
+		AC_SUBST([PVECLIB_DEFAULT_CFLAGS], [""]),
+	[AC_MSG_RESULT([no])]
+		AC_DEFINE([PVECLIB_DISABLE_DFP], [1], [Disable DFP])
+)
+
+if test "$retval" != "0"; then
+	AC_SUBST([PVECLIB_DEFAULT_CFLAGS],[POWER8_CFLAGS])
+fi
+
+##### Compiler default includes VSX #####
+## Use this test to set the minimum flags required to compile PVECLIB
+## This should cover BE compilers where the compiler default -mcpu is
+## not at least power7. Otherwise the compiler will disable VSX
+## features (like vector double) and associated built-ins.
+## VSX by the PowerISA architecture compliance rules this should
+## include
+
+CFLAGS=$AM_CFLAGS
+AC_MSG_CHECKING([if $CNAME default supports VSX])
+AC_LINK_IFELSE(
+    [AC_LANG_SOURCE([`cat testprograms/test_ppc_VSX.c`])],
+    [AC_MSG_RESULT([yes])]
+	AC_SUBST([PVECLIB_DEFAULT_CFLAGS], [""]),
+    [
+	AC_MSG_RESULT([no])
+	CFLAGS="$AM_CFLAGS $POWER7_CFLAGS"
+        AC_MSG_CHECKING([if $CNAME requires -mcpu=power7 for VSX])
+        AC_LINK_IFELSE(
+            [AC_LANG_SOURCE([`cat testprograms/test_ppc_VSX.c`])],
+            [AC_MSG_RESULT([yes])]
+                AC_SUBST([PVECLIB_DEFAULT_CFLAGS], [$POWER7_CFLAGS]),
+            [
+		AC_MSG_RESULT([no])
+                CFLAGS="$AM_CFLAGS $POWER8_CFLAGS"
+                AC_MSG_CHECKING([if $CNAME requires -mcpu=power8 for VSX])
+                AC_LINK_IFELSE(
+                    [AC_LANG_SOURCE([`cat testprograms/test_ppc_VSX.c`])],
+                    [AC_MSG_RESULT([yes])]
+                        AC_SUBST([PVECLIB_DEFAULT_CFLAGS], [$POWER8_CFLAGS]),
+		    [AC_MSG_RESULT([no])]
+			AC_DEFINE([PVECLIB_DISABLE_VSX], [1], [Disable VSX])
```

Comment:
So, sometimes `-mcpu=power7` will support VSX and sometimes not?
And sometimes `-mcpu=power8` will support VSX and sometimes not?
I guess that just depends on GCC having the support or not, right? Is there a minimum version of GCC supported by pveclib?
(No changes needed here, just checking.)

Reply:
Because LE does not support POWER7. -mcpu=power7 will fail so be bump to supported -mcpu=power8 target.